WikiWatershed/model-my-watershed
# Monitor: Add Contribute Data Modal

## Overview

Adds a Contribute Data modal instructing users to contribute their datasets to EnviroDIY and HydroShare. Also moves the filter button below the search bar, and adds a contribute data button there, as indicated in the mockups.

Connects #2815 

### Demo

![image](https://user-images.githubusercontent.com/1430060/40449975-3d786bac-5ea8-11e8-93d0-9bec207483f4.png)

![image](https://user-images.githubusercontent.com/1430060/40450006-5202d206-5ea8-11e8-8e57-710200856387.png)

![image](https://user-images.githubusercontent.com/1430060/40449987-44e17cbc-5ea8-11e8-83d0-869c0a599e7b.png)

Also tagging @jfrankl for visual review.

### Notes

The copy in the modals should be evaluated and finalized before merging. /cc @ajrobbins 

## Testing Instructions

* Check out this branch and `bundle --debug`
* Select a shape and proceed to Analyze
* Select Monitor tab. Ensure the Filter button is now below the search bar. Ensure it works as expected.
* Ensure there is a Contribute Data button, clicking which opens the modal shown above.
* Ensure clicking anywhere outside the modal, or the close button, closes the modal
* Ensure on repeated opening and closing there aren't multiple copies of the modal attached to the DOM, but only one.
* Ensure the links open in a new window, to the correct website, even on Safari.
